src/halmos/sevm.py

```diff
@@ -31,43 +31,65 @@
 
 Steps = Dict[int, Dict[str, Any]]  # execution tree
 
+
+# dynamic BitVecSort sizes
+class BitVecSortCache:
+    def __init__(self):
+        self.cache = {}
+
+    def __getitem__(self, size: int) -> BitVecSort:
+        if size not in self.cache:
+            self.cache[size] = BitVecSort(size)
+        return self.cache[size]
+
+
+BitVecSorts = BitVecSortCache()
+
+# known, fixed BitVecSort sizes
+BitVecSort1 = BitVecSorts[1]
+BitVecSort8 = BitVecSorts[8]
+BitVecSort160 = BitVecSorts[160]
+BitVecSort256 = BitVecSorts[256]
+BitVecSort264 = BitVecSorts[264]
+BitVecSort512 = BitVecSorts[512]
+
```

Comment:
Q: could you help me understand the behavior of this global mutable object when multiple SEVM objects are created?

for example, say we have two sevm objects:
```
from halmos.sevm import SEVM

sevm1 = SEVM(options)
sevm2 = SEVM(options)
```

will they have the same cache object, or a different one?

if the former, would it be safe when we run sevm1 and sevm2 in a thread or process pool?

in any case, can we avoid the use of global mutable objects?

Reply:
you're right I didn't think of that, both instances would access the same object because it's at the module level. I can make it read only to avoid this issue